### ISSUE TITLE
Position System and 3 Forces

### ISSUE THREAD

**Issue by Namek** [junkdog/artemis-odb#377]
I'm working on new `PositionSystem` that works on top of components such as:
1. `Force`
2. `Velocity` (that contains both acceleration and velocity,  maxSpeed and friction)
3. `Transform` - contains position, rotation, mostly. Let's forget that I have `desiredPos` and `currentPos` for collision resolving.

Force component is optional, because we can set `Velocity#acceleration`. If there's `Force`, it's copied to `Velocity#acceleration`.

Let's imagine a simple platformer like Mario.

Now, I'd like to have 3 forces for **convenience**:
1. `#force` - entity's own force that makes it run by it's own power (can be limited by `Velocity#maxSpeed`)
2. `#extForce` - some external force to kick the entity sometimes, e.g. pushed away after hit from shotgun (once) or just a big fan (constant air flow). Could be limited by `Velocity#maxExtSpeed`.
3. #gravity. Separated from `#extForce`, because if `#extForce` was too big, then summing up `#extForce` and `#gravity` AND limited by `Velocity#maxExtSpeed` could actually reduce gravity totally.

So, the logic for all those three forces is:
1. `#force` - `#acceleration`, then `#velocity` is calculated, optionally limited by `#maxSpeed`
2. `#extForce` - `#extAcceleration` then `#extVelocity` is calculated, velocity optionally limited by `#maxExtSpeed`
3. `#gravity` currently it's together with `2. #extForce` but I'd like to have it completely separated, because of velocity limiting

Position is being updated by all those 3 forces, of course.

So, basically, those 3 forces are *alike*. I can't make one `Force` component and add it 3 times. There's a second option: just create 3 different components but that's duplicating component code 3 times. What would you suggest?

**Comment by piotr-j** [comment]
Assuming contrib/1.0 with extended mappers
    Force force = mForce.create(entity);
    force.add(x, y);

I see no point in splitting forces like that, a force is a force, source is irrelevant. Limits should be done by stuff like damping, so constant move acceleration will result in constant speed, but it wont stop other forces to increase speed further.

**Comment by Namek** [comment]
It's just easier/quicker to set maximum speed, than calculating anti-force to achieve constant speed. As I said - looking for **convenience**.

**Comment by piotr-j** [comment]
Fine then
    
    Force force = mForce.create(entity);
    force.add(x, y);
    force.addExternal(x, y);
    force.addGravity(x, y);

I guess you can make a Force pojo and stick 3 of that in same component so you get limits and whatnot, delegate or not in the component.

**Comment by Namek** [comment]
In case if anyone was interested there's my current solution: https://github.com/Namek/teddy/blob/3d_refactor---master/core/src/net/namekdev/mgame/systems/base/PositionSystem.java

So I ended up with `Force` and `Gravity` components.
